R/tmiic.utils.R

```diff
@@ -0,0 +1,1343 @@
+    state_order$movavg = as.integer (state_order$movavg)
+    }
+  return (state_order)
+  }
+
+#-------------------------------------------------------------------------------
+# tmiic_check_parameters
+#-------------------------------------------------------------------------------
+# Checks on parameters for temporal mode
+#
+# As the temporal parameters n_layers, delta_t, movavg need to take different
+# values depending on the type (discrete/continuous) or contextual,
+# these parameters are moved in the state_order to have a value defined
+# for each variable (unless these information are already in the state_order,
+# in such case, the parameters are ignored).
+# The other temporal parameters, having one value not tuned per variable
+# are added in the list of parameters.
+#
+# Params:
+# - state_order: the dataframe  returned by tmiic_check_state_order_part1
+# - params: the list of parameters (used only to add temporal parameters)
+# - all possible temporal parameters of miic method
+# Returns: a list with 2 items:
+# - state_order: the state_order, with temporal parameters eventually added
+# - params: the list of parameters with temporal parameters added
+#-------------------------------------------------------------------------------
+tmiic_check_parameters <- function (state_order, params,
+  n_layers, delta_t, movavg, keep_max_data, max_nodes)
+  {
+  # Check number of layers parameter
+  #
+  if ( ! is.null (n_layers) )
+    {
+    if ( test_param_wrong_int (n_layers, min=2, max=NA) )
+      {
+      if ( "n_layers" %in% colnames(state_order) )
+        miic_warning ("parameters", "supplied value ", list_to_str (n_layers),
+          " for the number of layers is invalid,",
+          " if not NULL, it must be an integer >= 2.",
+          " This issue has no impact as the number of layers is provided",
+          " in the state_order.")
+      else
+        miic_warning ("parameters", "supplied value ", list_to_str (n_layers),
+          " for the number of layers is invalid,",
+          " if not NULL, it must be an integer >= 2.",
+          " The number of layers will be estimated from the data.")
+      }
+    else # valid n_layers
+      {
+      if ( ! ("n_layers" %in% colnames(state_order)) )
+        {
+        state_order$n_layers = n_layers
+        state_order$n_layers[state_order$is_contextual == 1] = 1
+        }
+      else # n_layers in state_order
+        {
+        na_in_so = is.na (state_order$n_layers)
+        if ( any (na_in_so) )
+          {
+          miic_warning ("parameters", "the number of layers is supplied both",
+            " in the state_order and as parameter. As some values are missing",
+            " in the state_order, the parameter will be used to fill these",
+            " missing values.")
+          state_order$n_layers[na_in_so & (state_order$is_contextual == 0)] = n_layers
+          }
+        else
+          miic_warning ("parameters", "the number of layers is supplied both",
```

Comment:
EN (OD)

Reply:
is both supplied in the state_order and as a parameter.